### ISSUE TITLE
hello，I meet some error in this code when input black mask

### ISSUE THREAD

**Issue by zouzhekang** [HighwayWu/FOCAL#3]
Hi, I run a real video with your weights and I get messy results, even if I retrain it. And if the input training set mask is all black, this code will directly report an error, can you reply me? pleeeeeeease

你好，我用你的权重跑一个真实的视频会输出乱七八糟的结果，即使我重新训一次也是这个结果。而且输入的训练集掩码如果是全黑的话这个代码就直接报错了，请问是怎么回事呢？

**Comment by HighwayWu** [comment]
Thanks for your asking.
1) This work aims at detecting forgery image (video frame is also okay). If you retrain it with real data, the objective loss cannot be reasonably calculated.
2) There is an assertion in line [188](https://github.com/HighwayWu/FOCAL/blob/8866db2b4871e9fe666faa272f287babef84fbc8/main.py#L188) to determine if the input mask is all black, and there should be no error. May I ask what the specific error message is?

**Comment by zouzhekang** [comment]
> Thanks for your asking.
> 
> 1. This work aims at detecting forgery image (video frame is also okay). If you retrain it with real data, the objective loss cannot be reasonably calculated.
> 2. There is an assertion in line [188](https://github.com/HighwayWu/FOCAL/blob/8866db2b4871e9fe666faa272f287babef84fbc8/main.py#L188) to determine if the input mask is all black, and there should be no error. May I ask what the specific error message is?

yes, I saw it, but an error will still be reported when all masks in a batch are black images. 
I solved it by using the try ... except ...
But I put in the real video and there's still an output, I think this may be caused by data imbalance, no all black masks participated in the training.

**Comment by HighwayWu** [comment]
Sure. There should be at least one fake mask in each batch, otherwise the loss cannot be calculated (actually empty).

**Comment by zouzhekang** [comment]
> Sure. There should be at least one fake mask in each batch, otherwise the loss cannot be calculated (actually empty).

OK！Thank you for your reply.
